src/main/java/TaskList.java

```diff
@@ -0,0 +1,88 @@
+import java.io.IOException;
+import java.io.File;
+import java.io.BufferedReader;
+import java.io.FileReader;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class TaskList {
+    //abstract the arraylist access and parsing of file
+    private ArrayList<Task> taskList;
+
+    TaskList(File file) throws IOException {
+        BufferedReader br = new BufferedReader(new FileReader(file));
+        String line = br.readLine();
+        taskList = new ArrayList<>();
+        while (line != null) {
+            String[] split = line.split("%d%");
+            if (split.length > 3) {
+                taskList.add(Task.createTask(split[0], split[1], split[2], split[3]));
+            } else {
```

Comment:
Can add an additional else statement to capture everything else. For easier debugging.